Automattic/pocket-casts-ios
# Issue 12 – Replace MaterialBottomSheet

Fixes #12 [Replace MaterialBottomSheet with our own implementation](https://github.com/Automattic/pocket-casts-ios/issues/12)

This PR replaces usages of `MaterialComponents/BottomSheet` with Apple's [UISheetPresentationController](https://developer.apple.com/documentation/uikit/uisheetpresentationcontroller) and removes the 3rd-party dependency.

## To test

The steps for testing will be different for each sheet in the app. These are the steps to test the Sleep Timer for anyone that would like to preview the new sheet.

1. Tap a Podcast to enter the list of episodes.
2. Tap an episode from the list to start playing one of the podcast episodes.
3. Tap the "now playing" footer at the bottom to go to the fullscreen detail for the podcast episode.
4. Tap the 💤 button in the bottom footer. The Sleep Timer bottom sheet will appear in the new bottom sheet.

## Checklist

- [x] I have considered adding unit tests for my changes. (UI changes, no unit tests)

## Before & After

Note that a video is included for the Overflow Actions behavior as, when the user taps the Edit button, that sheet enlarges and becomes undismissible via swipe actions.

| **App Scene** | **Before** | **After** |
|-----------|---------|---------|
| **Effects Sheet** | ![effects_sheet_before](https://github.com/Automattic/pocket-casts-ios/assets/4069415/396c00b9-0268-4d74-8214-fbcbc3e78a39) | ![effects_sheet_after](https://github.com/Automattic/pocket-casts-ios/assets/4069415/3bf03f01-51da-4ff4-8f5b-e1007bbd1b76) |
| **Multiselect Actions** | ![multiselect_sheet_before](https://github.com/Automattic/pocket-casts-ios/assets/4069415/10c6a538-da90-426c-b5ba-94bad152c30c) | ![multiselect_sheet_after](https://github.com/Automattic/pocket-casts-ios/assets/4069415/4d97787a-1701-41a9-8341-2fe341f465d1) |
| **Multiselect Actions (Expanded)** | ![multiselect_sheet_expanded_before](https://github.com/Automattic/pocket-casts-ios/assets/4069415/b3e885dd-5f5b-48b2-8a39-2aebf60224a7) | ![multiselect_sheet_expanded_after](https://github.com/Automattic/pocket-casts-ios/assets/4069415/3c575338-87be-459d-afd2-7db032e959f9) |
| **Overflow Actions** | ![overflow_shelf_actions_before](https://github.com/Automattic/pocket-casts-ios/assets/4069415/028bf051-9477-4b11-92b7-42d31712bdd8) | ![overflow_shelf_actions_after](https://github.com/Automattic/pocket-casts-ios/assets/4069415/1d03f1c0-539c-48a2-b563-c57caec6fe8f) |
| **Overflow Actions Edit Behavior** | <video src="https://github.com/Automattic/pocket-casts-ios/assets/4069415/c1e01ce9-a90e-4fa9-95ab-ea2bfbdd8012"/> | <video src="https://github.com/Automattic/pocket-casts-ios/assets/4069415/8a9d1f96-998e-40ff-a759-66903f5b4eb8"/> |
| **SwiftUI Wrapper Example (EoY)** | ![presentModally(year_end_example)_sheet_before](https://github.com/Automattic/pocket-casts-ios/assets/4069415/0d9d9232-c04a-4313-bd77-0320d2d81ffc) | ![presentModally(year_end_example)_sheet_after](https://github.com/Automattic/pocket-casts-ios/assets/4069415/e583588b-30ba-46ac-b592-748a68c5c7fa) |
| **Profile** | ![profile_sheet_before](https://github.com/Automattic/pocket-casts-ios/assets/4069415/602fded1-72d4-4b4b-97e9-42fbefd50b29) | ![profile_sheet_after](https://github.com/Automattic/pocket-casts-ios/assets/4069415/772e67e8-0b07-4892-b5a6-268ebb2b4e5b) |
| **Sleep Timer** | ![sleep_timer_before](https://github.com/Automattic/pocket-casts-ios/assets/4069415/b1095b5e-c70c-46a3-922a-3b6807ab7754) | ![sleep_timer_after](https://github.com/Automattic/pocket-casts-ios/assets/4069415/9ed92e37-d632-47c0-80c3-272a2ea57e64) |
